csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/AssemblyCache.cs

```diff
@@ -20,44 +19,16 @@ internal class AssemblyCache
         /// assembly cache.
         /// </param>
         /// <param name="logger">Callback for progress.</param>
-        public AssemblyCache(IEnumerable<string> paths, IEnumerable<string> frameworkPaths, ILogger logger)
+        public AssemblyCache(IEnumerable<AssemblyPath> paths, IEnumerable<string> frameworkPaths, ILogger logger)
         {
             this.logger = logger;
             foreach (var path in paths)
             {
-                if (File.Exists(path))
-                {
-                    dllsToIndex.Add(path);
-                    continue;
-                }
-
-                if (Directory.Exists(path))
-                {
-                    logger.LogInfo($"Finding reference DLLs in {path}...");
-                    AddReferenceDirectory(path);
-                }
-                else
-                {
-                    logger.LogInfo("AssemblyCache: Path not found: " + path);
-                }
+                path.Process(dllsToIndex, logger);
```

Comment:
This `Process` call could be changed to simply return a list of DLL paths, and then the return value could be used: `dllsToIndex.AddRange(path.GetFilePaths(logger))`. This way the `AssemblyPath` is more self contained as we're not pushing "state" into it.